OpenThaiGPT/openthaigpt-pretraining
# build(all): add formatter, linter, precommit hooks, github action check

This PR adds tons of static code checkers:
- MyPy for type checking
- Flake8 for styles guide
- Black for code formatting

Checker will run on the following conditions:
- automatically format/lint if you are using VS Code
- committing files in Git
- When opening PR

To test this PR for commit hooks do:
- checkout to `new/formater` branch
- Run `pip install -r requirements.txt` and `pre-commit install`
- Edit/Create any python file and stage changes
- Try to commit. At this point **we should have all checkers running against your staged files**

Close #71
